public/app/features/correlations/Forms/QueryEditorField.tsx

```diff
@@ -23,8 +50,52 @@ export const QueryEditorField = ({ dsUid, invalid, error, name }: Props) => {
     }
     return getDataSourceSrv().get(dsUid);
   }, [dsUid]);
+
   const QueryEditor = datasource?.components?.QueryEditor;
 
+  const handleValidation = (value: DataQuery) => {
+    const interval = '1s';
+    const intervalMs = 1000;
+    const id = generateKey();
+    const queries = [{ ...value, refId: 'A' }];
+
+    const transaction: QueryTransaction = {
+      queries,
+      request: {
+        app: CoreApp.Unknown,
+        timezone: 'utc',
+        startTime: Date.now(),
+        interval,
+        intervalMs,
+        targets: queries,
+        range: getDefaultTimeRange(),
+        requestId: 'correlations_' + id,
+        scopedVars: {
+          __interval: { text: interval, value: interval },
+          __interval_ms: { text: intervalMs, value: intervalMs },
+        },
+      },
+      id,
+      done: false,
+    };
+
+    if (datasource) {
+      runRequest(datasource, transaction.request).subscribe((panelData) => {
+        if (!panelData || panelData.state === 'Error' || panelData.series.length === 0) {
```

Comment:
When checking for number of series we also need to check the status - otherwise it will flash "not valid" message during Loading state.

```suggestion
if (
          !panelData ||
          panelData.state === 'Error' ||
          (panelData.state === 'Done' && panelData.series.length === 0)
        ) {
```